src/ucc/api/ucc.h

```diff
@@ -2220,5 +2220,53 @@ ucc_status_t ucc_ee_wait(ucc_ee_h ee, ucc_ev_t *ev);
  */
 ucc_status_t ucc_collective_triggered_post(ucc_ee_h ee, ucc_ev_t *ee_event);
 
+typedef void *ucc_mem_map_mem_h;
+
+/**
+ * @ingroup UCC_DATATYPE
+ */
+typedef enum {
+    UCC_MEM_MAP_EXPORT = 0, /*!< Indicate ucc_mem_map() should export
+                                 memory handles from TLs used by context */
+    UCC_MEM_MAP_IMPORT = 1  /*!< Indicate ucc_mem_map() should import
+                                memory handles from user memory handle */
+} ucc_mem_map_flags_t;
+
+/**
+ * @ingroup UCC_CONTEXT
+ * @brief Routine registers or maps memory for use in future collectives.
+ *
+ * This routine maps a user-specified memory segment with a ucc_context_h. The
+ * segment is considered "mapped" with the context until either the user calls
+ * ucc_mem_unmap or ucc_context_destroy(). A handle to the mapped
```

Comment:
I think the problem relates to whether mapped memory is associated with a context or its own object. From my understanding, mapped memory would be used by collectives for communication purposes, meaning it would need to be associated with a context as a context contains network resources. In that case, when a user decides to destroy a context, the memory must be unmapped in order to prevent erroneous usage afterwards.  @manjugv please correct me.

Reply:
I will update the text to make it the user's responsibility